### PR TITLE
feat(atlassian): add atlassian oauth2 provider

### PR DESCRIPTION
**Pull request by Ahuahuachi**
- **feat(atlassian): add atlassian oauth2 provider**
- **chore(atlassian): add atlassian provider documentation**

# Submitting Pull Requests

## General

 - [x] Make sure you use [semantic commit messages](https://seesparkbox.com/foundry/semantic_commit_messages).
       Examples: `"fix(google): Fixed foobar bug"`, `"feat(accounts): Added foobar feature"`.
 - [x] All Python code must formatted using Black, and clean from pep8 and isort issues.
 - [x] JavaScript code should adhere to [StandardJS](https://standardjs.com).
 - [x] If your changes are significant, please update `ChangeLog.rst`.
 - [x] If your change is substantial, feel free to add yourself to `AUTHORS`.

 ## Provider Specifics

 In case you add a new provider:

- [x] Make sure unit tests are available.
- [x] Add an entry of your provider in `test_settings.py::INSTALLED_APPS` and `docs/installation.rst::INSTALLED_APPS`.
- [x] Add documentation to `docs/providers/<provider name>.rst` and `docs/providers/index.rst` Provider Specifics toctree.
- [x] Add an entry to the list of supported providers over at `docs/overview.rst`.

### CODE REVIEW

**Comment by pennersr** [comment]
Question: Isn't Atlassian a fully OpenID Connect compliant provider? Because if it is, we should be able to just use the existing `openid_connect` provider out of the box without adding a new dedicated/hardcoded provider.

**Comment by Ahuahuachi** [comment]
AFAIK the OpenID compliant products of Atlassian are the On-Premise products (ex. Jira Server). The Cloud Services (ex. Jira Cloud) are not.

File: allauth/socialaccount/providers/atlassian/provider.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    def extract_uid(self, data):
+        return data["account_id"]
+
+    def extract_common_fields(self, data):
+        return {
+            "email": data.get("email"),
+            "name": data.get("name"),
+            "username": data.get("nickname"),
+            "email_verified": data.get("email_verified"),
```
Comment (pennersr) [inline allauth/socialaccount/providers/atlassian/provider.py:26]: **Question**: I don't see this field being documented, nor is it present in your test response. Does this property exist?

https://developer.atlassian.com/cloud/jira/platform/oauth-2-3lo-apps/#how-do-i-retrieve-the-public-profile-of-the-authenticated-user-

File: allauth/socialaccount/providers/atlassian/views.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    def complete_login(self, request, app, access_token, **kwargs):
+        headers = {
+            "Authorization": f"Bearer {access_token}",
+            "Accept": "application/json",
+        }
+        response = (
+            get_adapter().get_requests_session().get(self.profile_url, headers=headers)
+        )
+        data = response.json()
```
Comment (pennersr) [inline allauth/socialaccount/providers/atlassian/views.py:23]: Can you add a `response.raise_for_status()` before calling `.json()`?

**Review by pennersr — CHANGES_REQUESTED** [review]

File: allauth/socialaccount/providers/atlassian/provider.py (hunk 1 above)
Comment (Ahuahuachi) [reply allauth/socialaccount/providers/atlassian/provider.py:26]: Sorry, my bad. I didn't checked the example on the official documentation with the actual call.
The field exists, so added to the test response.

**Pull request merged**